src/pages/workspaces/workspace/jobHistory/SubmissionDetails.js

```diff
@@ -160,10 +161,10 @@ const SubmissionDetails = _.flow(
       div({ style: { display: 'grid', gridTemplateColumns: '1fr 4fr' } }, [
         div({ style: { display: 'grid', gridTemplateRows: '1fr auto' } }, [
           makeSection('Workflow Statuses', [
-            succeeded && makeStatusLine(successIcon, `Succeeded: ${succeeded.length}`, { marginTop: '0.5rem' }),
-            failed && makeStatusLine(failedIcon, `Failed: ${failed.length}`, { marginTop: '0.5rem' }),
-            running && makeStatusLine(runningIcon, `Running: ${running.length}`, { marginTop: '0.5rem' }),
-            submitted && makeStatusLine(submittedIcon, `Submitted: ${submitted.length}`, { marginTop: '0.5rem' })
+            succeeded && makeStatusLine(statusType.succeeded.icon, addCountSuffix(statusType.succeeded.label(), succeeded.length), { marginTop: '0.5rem' }),
+            failed && makeStatusLine(statusType.failed.icon, addCountSuffix(statusType.failed.label(), failed.length), { marginTop: '0.5rem' }),
+            running && makeStatusLine(statusType.running.icon, addCountSuffix(statusType.running.label(), running.length), { marginTop: '0.5rem' }),
+            submitted && makeStatusLine(statusType.submitted.icon, addCountSuffix(statusType.submitted.label(), submitted.length), { marginTop: '0.5rem' })
```

Comment:
Similarly here, if you did not decompose the groups:
```javascript
'succeeded failed running submitted'.split(' ').filter(s => groups[s])
  .map(s =>
  makeStatusLine(statusType[s].icon, addCountSuffix(statusType[s].label(), groups[s].length)))
```

Reply:
Thank you! This simplifies the code quite a bit.